Add accessors for TM columnForCell and cellsForColumn

SaganBolliger opened the pull request:
Fixes #839 

@chetan51

chetan51 commented:
:+1:

SaganBolliger commented:
@chetan51 Looks like travis timed out for one of its builds and AV was cancelled. Can you restart them?

chetan51 commented:
@SaganBolliger I can't restart them. Can you push an empty commit to restart the build?

rhyolight commented:
One of the travis linux builds had a test failure, which is weird. I restarted it. I'm trying to figure out why the AV builds are not running.

rhyolight commented:
Forced AV re-build...

SaganBolliger commented:
@rhyolight That test failure is on a test I wrote. It checks that the python bindings are importable. The fact that it is failing likely indicates that was an issue with the build further back. I can't see any errors in the build log though. Very strange.

SaganBolliger commented:
@rhyolight I think I found it in the [raw log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/105298155/log.txt)
```
  Running setup.py install for pycapnp
    cc -c /tmp/timer_createCCznuK.c -o /tmp/tmpsCej7r/tmp/timer_createCCznuK.o
    cc /tmp/tmpsCej7r/tmp/timer_createCCznuK.o -o /tmp/tmpsCej7r/a.out
    /tmp/tmpsCej7r/tmp/timer_createCCznuK.o: In function `main':
    timer_createCCznuK.c:(.text+0x15): undefined reference to `timer_create'
    collect2: ld returned 1 exit status
    cc -c /tmp/tmpsCej7r/vers.cpp -o tmp/tmpsCej7r/vers.o --std=c++11
    cc: error: unrecognized option â€˜--std=c++11â€™
```
Has the configuration for the gcc build changed such that it doesn't support C++11 anymore?

rhyolight commented:
@SaganBolliger Certainly not. I suggest you install Ubuntu in a VM and debug there with GCC.

scottpurdy commented:
@oxtopus - Can you help @SaganBolliger with this if he is still having issues tomorrow (I'm on PTO tomorrow).

SaganBolliger commented:
Looks like my latest commit fixes the issue with gcc. It seems like the issue was that using gcc, swig was unable to handle a templated C++ method. However the build failed silently producing no errors, and it was only evident that it had failed once you tried importing the python bindings.

The change I've made simultaneously fixes an API inconsistency in that `cellsForColumn` is used as part of the public API, but it returns Cell objects, whereas the API convention is to return cell indices. I've changed it to return indices instead and defined a new method `cellsForColumnCell` that returns Cell objects for internal use. Updated callers as necessary.

@chetan51 Please review.

chetan51 commented:
:+1:
